### ISSUE TITLE
Simplification of declarations

### ISSUE THREAD

**Issue by chriseth** [powdr-labs/powdr#532]
It would be nice if we could simplify the declarations. If possible, there should only be one declaration statement, what we declare follows from the type, or maybe some markers.

Some examples (first line is old syntax, second line is new syntax):
```rust
// constant definition
constant %N = 16;
let N = 16;

// constant definition
constant %last_row = %N - 1;
let last_row = N - 1;

// constant polynomial definition
col fixed islast(i) { ... }
let islast(i) { .... }

// constant poly definition
col fixed even = [0, 1]*;
let even = [0, 1]*;

// witness declaration
col witness X;
witness X; // could also just say "let X"

// macro definition
macro ite(C, A, B) { is_nonzero(C) * A + is_zero(C) * B};
let ite(C, A, B) { is_nonzero(C) * A + is_zero(C) * B};

```

It might be a bit weird that we don't distinguish between witness and fixed polys, but maybe that just needs to getting used to?

We could have different syntax for `let` and `fn`.

**Comment by chriseth** [comment]
We could also say that functions are defined using
```rust
let islast = |i| { ... }
```

**Comment by chriseth** [comment]
This system could also improve macros: Currently, macros can return a list of constraints and a single expression (depending on whether you use `,` or `;`). This is weird.

If we type everything properly, we can check if the macro function returns a constraint (or a list of constraints) or if it returns an expression. At the pil level, only constraints or lists of constraints are allowed.

**Comment by chriseth** [comment]
This is kind of implemented already in #564
